fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java

```diff
@@ -45,11 +48,31 @@ public class CheckAfterRewrite extends OneAnalysisRuleFactory {
     public Rule build() {
         return any().then(plan -> {
             checkAllSlotReferenceFromChildren(plan);
+            checkNoUnbounded(plan);
             checkMetricTypeIsUsedCorrectly(plan);
             return null;
         }).toRule(RuleType.CHECK_ANALYSIS);
     }
 
+    private void checkNoUnbounded(Plan plan) {
+        List<Expression> unboundedExprs = Lists.newArrayList();
+        boolean unbounded = plan.getExpressions().stream().anyMatch(
+                expr -> {
+                    if (expr.getInputSlots().stream().anyMatch(UnboundSlot.class::isInstance)) {
+                        unboundedExprs.add(expr);
+                        return true;
+                    } else if (expr.getInputSlots().stream().anyMatch(UnboundFunction.class::isInstance)) {
+                        unboundedExprs.add(expr);
+                        return true;
+                    }
+                    return false;
```

Comment:
both `UnboundSlot` and `UnboundFunction` are implement `Unbound`.
just need to match `Unbound`

Reply:
done